src/Microsoft.Data.SqlClient/tests/FunctionalTests/SslOverTdsStreamTest.cs

```diff
@@ -0,0 +1,347 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.Tests
+{
+    public static class SslOverTdsStreamTest
+    {
+        [Theory]
+        [InlineData(0),InlineData(3),InlineData(128), InlineData(2048), InlineData(8192)]
+        public static void ReadWrite(int readLimit)
+        {
+            const int EncaspulatedPacketCount = 4;
```

Comment:
Can you move these hardcoded items to InlineData as well?

Reply:
I could but there is no benefit to doing so which is why I chose to leave them hardcoded. The choice of multiple packets is already doing more than the real-world usage will require. The pattern of data is chosen so that it can't accidentally collide (well, it can but it's numerically impractical to do so and well outside the scope of these tests). 

Do you want me to add the additional complexity?

Reply:
Split out the items into the TheoryData generator. If for some reason anyone every wants more all the have to do is add more cases to the initialization property function.